src/screens/nano/NanoContractDetail.js

```diff
@@ -0,0 +1,203 @@
+    e.preventDefault();
+    context.showModal(MODAL_TYPES.NANOCONTRACT_CHANGE_ADDRESS, {
+      nanoContractID: ncId
+    });
+  }
+
+  const loadData = async () => {
+    await loadBlueprintInformation();
+    await loadNCData();
+  }
+
+  const loadBlueprintInformation = async () => {
+    if (blueprintInformationAux) {
+      return;
+    }
+
+    setLoading(true);
+    try {
+      const blueprintInformationResponse = await hathorLib.ncApi.getBlueprintInformation(nc.blueprintId);
+      // We need this blueprint information response to call the following get state
+      // The set state is not sync, so we need to store it in a common variable to be used in the next call
+      blueprintInformationAux = blueprintInformationResponse;
+      setBlueprintInformation(blueprintInformationResponse);
+      // Store in redux, so it can be reused by other nano contracts
+      dispatch(addBlueprintInformation(blueprintInformationResponse));
+    } catch(e) {
+      // Error in request
+      setLoading(false);
+      setErrorMessage(t`Error getting blueprint details.`);
+    };
+  }
+
+  const loadNCData = async () => {
+    setLoading(true);
+    setData(null);
+    try {
+      const state = await hathorLib.ncApi.getNanoContractState(ncId, Object.keys(blueprintInformationAux.attributes), [], []);
+      setLoading(false);
+      setData(state);
+    } catch(e) {
+      // Error in request
+      setLoading(false);
+      setErrorMessage(t`Error getting nano contract state.`);
+    };
+  }
+
+  const renderBody = () => {
+    if (loading) {
+      return <ReactLoading type='spin' color={colors.purpleHathor} delay={500} />;
+    }
+
+    if (errorMessage) {
+      return <p className='text-danger mb-4'>{errorMessage}</p>;
+    }
+
+    return renderNCData();
+  }
+
+  const renderNanoAttributes = () => {
+    return Object.keys(data.fields).map((field) => {
+      const value = get(data.fields[field], 'value', undefined);
+      if (value === undefined) {
+        // Error getting field
+        // Since we are using the attributes from the blueprint information API to
+        // get the state, we know that the fields exist. If value is undefined, it
+        // means they are dict fields, which we should just show the types of them for now
+        const type = get(blueprintInformation.attributes, field);
+        return <p key={field}><strong>{field}: </strong>{type}</p>;
+      }
+
+      // Some fields should be better parsed, e.g., timestamp, address
+      // however we don't have a simple and generic way to knowing it
+      // this was discussed and we will have this in the future, so
+      // for now we keep this UI and when we have this feature in the
+      // hathor-core, we can improve the user UI
+      return <p key={field}><strong>{field}: </strong>{value === null ? ' - ' : value}</p>;
+    });
+  }
+
+  const renderNanoMethods = () => {
+    return Object.keys(blueprintInformation.public_methods).filter((method) =>
```

Comment:
Use lodash get with a default since `public_methods` can be null or undefined, making this throw.

Reply:
Another alternative would be `Object.keys(blueprintInformation.public_methods || {})`

Reply:
It will never be null/undefined. The API returns an empty object `{}` in case there are no public methods.

Reply:
If everything works as intended it should not be null/undefined but we should consider this case and either warn the user with an error or quietly discard it with an empty object default.

Reply:
Makes sense, added the protection https://github.com/HathorNetwork/hathor-wallet/pull/623/commits/b7a84c1b4326caf4b93daa07aeda4422dda2bc2d